### ISSUE TITLE
Not all devices are getting discovered

### ISSUE THREAD

**Issue by raman325** [home-assistant/devices-experiment#29]
I ran into another issue with `via_device` where the `via_device` device was not included in the device list. This is happening because the template will only return devices that have entities attached to them. As a result, for hub integrations, you may miss the hubs themselves.

**Comment by balloob** [comment]
I feel like we should instead get a method to just iterate all devices instead of the back this template does. We have same for areas etc

**Comment by raman325** [comment]
I agree that having a proper export/websocket integration would be better for the device database. But I think this template would be beneficial in general. Example would be to heal every node in a zwave network. Whats your concern with this?

This may not be the right implementation so happy to adjust scope/design based on the convo. Btw this is the same logic reapplied from integration_entities

**Comment by balloob** [comment]
The concern is that bdraco had mentioned the other day that `integration_entities` is always high up on the profiling list as an expensive function, because of the way it finds integrations. It's also possible that it breaks because integration titles are not unique.
